WooCommerce/WooCommerceTests/System/WooConstantsTests.swift

```diff
@@ -0,0 +1,18 @@
+import XCTest
+@testable import WooCommerce
+
+final class WooConstantsTests: XCTestCase {
+
+    func testAllThrustedURLsAreValid() {
+        // Given
+        let allTrustedURLPaths = WooConstants.URLs.allCases
+
+        // When
+        let allTrustedURLs = allTrustedURLPaths.map { $0.asURL() }
+
+        // Then
+        zip(allTrustedURLPaths, allTrustedURLs).forEach { (path, url) in
```

Comment:
A problem when one of the URLs is malformed is that a `fatalError` will happen and **all the unit tests** will be canceled. It's not a good experience for us if it happens. It's quite unlikely though so I think it's fine. 

Or maybe we could change the unit test to do the `URL(string: )` call itself. But I guess that defeats the purpose. 

It's fine. I'm over-analyzing this. 😄

Reply:
Yeah, I know this isn't ideal.
One potential solution would be to add some kind of [catchable fatal error](https://medium.com/@marcosantadev/how-to-test-fatalerror-in-swift-e1be9ff11a29) to the project. 

What do you think?

Reply:
I'll read that and get back to you. 🙂

Reply:
@Ecarrion I just read it and it looks like a neat idea! I'm not so sure about overriding `fatalError()` itself as that may be very surprising for new developers. Maybe we can make our own global function? We'd also have to think about how we can make it reusable across the projects. 

There's an issue that's somewhat related to this: https://github.com/woocommerce/woocommerce-ios/issues/2364